FWCore/Sources/interface/VectorInputSourceDescription.h

```diff
@@ -0,0 +1,30 @@
+#ifndef FWCore_Sources_VectorInputSourceDescription_h
+#define FWCore_Sources_VectorInputSourceDescription_h
+
+/*----------------------------------------------------------------------
+
+VectorInputSourceDescription : the stuff that is needed to configure
+a VectorinputSource that does not come in through the ParameterSet  
+----------------------------------------------------------------------*/
+
+#include "FWCore/Framework/src/PreallocationConfiguration.h"
+
+namespace edm {
+  class PreallocationConfiguration;
+  class ProductRegistry;
+
+  struct VectorInputSourceDescription {
+    VectorInputSourceDescription() :
+      productRegistry_(nullptr) {
+    }
+
+    VectorInputSourceDescription(ProductRegistry& preg, PreallocationConfiguration const& allocations) :
+      productRegistry_(&preg), allocations_(&allocations) {
+    }
+
+    ProductRegistry* productRegistry_;
```

Comment:
Would it make more sense to use ```std::shared_ptr<ProductRegistry>``` here and then just have the ```VectorInputSource``` do a copy? The reason is this piece of code actually knows what is the proper ownership rules for the object.